src/createAnimatedComponent.tsx

```diff
@@ -580,16 +581,27 @@ export default function createAnimatedComponent(
 
     _filterNonAnimatedStyle(inputStyle: StyleProps) {
       const style: StyleProps = {};
+      let changed = false;
       for (const key in inputStyle) {
         const value = inputStyle[key];
         if (!hasAnimatedNodes(value)) {
           style[key] = value;
+          changed = changed || style[key] !== this._lastSentStyle[key];
         } else if (value instanceof AnimatedValue) {
           // if any style in animated component is set directly to the `Value` we set those styles to the first value of `Value` node in order
           // to avoid flash of default styles when `Value` is being asynchrounously sent via bridge and initialized in the native side.
           style[key] = value._startingValue;
+          changed = changed || style[key] !== this._lastSentStyle[key];
         }
       }
+      if (!changed) {
+        const lastKeys = Object.keys(this._lastSentStyle);
+        const inputKeys = Object.keys(inputStyle);
+        if (lastKeys.every((k) => inputKeys.includes(k))) {
```

Comment:
Not sure if it is safe enough to test only for the lengths. The case that bothers me is:
```
inputStyle = {
  var1: 1,
  var2: 2,
  var3: undefined,
}
lastStyle = {
   var1: 1,
   var2: 2,
   var4: 5,
}
```

```